code/modules/library/lib_items.dm

```diff
@@ -18,106 +18,123 @@
 		anchored = !anchored
+		playsound(get_turf(src), 'sound/items/Ratchet.ogg', 50, 1)
+		user.visible_message("<span class='warning'>[user] [anchored ? "":"un"]anchors \the [src] [anchored ? "to":"from"] the floor.</span>", \
+		"<span class='notice'>You [anchored ? "":"un"]anchor the [src] [anchored ? "to":"from"] the floor.</span>")
 	else if(istype(O, /obj/item/weapon/pen))
-		var/newname = stripped_input(usr, "What would you like to title this bookshelf?")
+		var/newname = stripped_input(user, "What category title would you like to give to this [name]?")
 		if(!newname)
 			return
 		else
 			name = ("bookcase ([sanitize(newname)])")
+	else if(O.damtype == BRUTE || O.damtype == BURN)
+		user.delayNextAttack(10) //We are attacking the bookshelf
+		health -= O.force
+		user.visible_message("<span class='warning'>\The [user] hits \the [src] with \the [O].</span>", \
+		"<span class='warning'>You hit \the [src] with \the [O].</span>")
+		healthcheck()
+		return
+
 	else
-		switch(O.damtype)
-			if("fire")
-				src.health -= O.force * 1
-			if("brute")
-				src.health -= O.force * 0.75
-			else
-		if (src.health <= 0)
-			visible_message("<span class=warning>The bookcase is smashed apart!</span>")
-			getFromPool(/obj/item/stack/sheet/wood, get_turf(src), 3)
-			qdel(src)
-		..()
+		..() //Weapon checks for weapons without brute or burn damage type and grab check
 
 /obj/structure/bookcase/attack_hand(var/mob/user as mob)
 	if(contents.len)
-		var/obj/item/weapon/book/choice = input("Which book would you like to remove from the shelf?") as null|obj in contents
+		var/obj/item/weapon/book/choice = input("Which book would you like to remove from \the [src]?") as null|obj in contents
 		if(choice)
-			if(!usr.canmove || usr.stat || usr.restrained() || !in_range(loc, usr))
+			if(user.restrained() || user.stat || user.weakened || user.stunned || user.paralysis || user.resting || get_dist(user, src) > 1)
 				return
-			if(ishuman(user))
-				if(!user.get_active_hand())
-					user.put_in_hands(choice)
+			if(!user.get_active_hand())
+				user.put_in_hands(choice)
 			else
-				choice.loc = get_turf(src)
+				choice.forceMove(get_turf(src))
 			update_icon()
 
 /obj/structure/bookcase/ex_act(severity)
 	switch(severity)
 		if(1.0)
-			for(var/obj/item/weapon/book/b in contents)
-				qdel(b)
-			qdel(src)
+			for(var/obj/item/I in contents)
+				qdel(I)
+			Destroy(src)
 			return
 		if(2.0)
-			for(var/obj/item/weapon/book/b in contents)
-				if (prob(50)) b.loc = (get_turf(src))
-				else qdel(b)
-			qdel(src)
+			for(var/obj/item/I in contents)
+				if(prob(50))
+					qdel(I)
+			Destroy(src)
 			return
 		if(3.0)
-			if (prob(50))
-				for(var/obj/item/weapon/book/b in contents)
-					b.loc = (get_turf(src))
-				qdel(src)
+			if(prob(50))
+				Destroy(src)
 			return
-		else
 	return
 
+/obj/structure/bookcase/Destroy()
+
+	density = 0 //Sanity while we do the rest

```

Comment:
The parent for destroy already does all this shit anyway, invisibility density whatnot I think